### ISSUE TITLE
BUG: Website too large to deploy

### ISSUE THREAD

**Issue by hoechenberger** [mne-tools/mne-bids-pipeline#752]
It's time to cut a new release, this is the auto-generated changelog from GH:

> ## What's Changed
> * MAINT: Use a large resource class to speed up build by @larsoner in https://github.com/mne-tools/mne-bids-pipeline/pull/741
> * BUG: Fix bug with mf_reference_run by @larsoner in https://github.com/mne-tools/mne-bids-pipeline/pull/742
> * Fix bug with too many JSON files found during empty room matching by @allermat in https://github.com/mne-tools/mne-bids-pipeline/pull/743
> * Fix documentation examples for ch_types configuration option by @allermat in https://github.com/mne-tools/mne-bids-pipeline/pull/745
> * MAINT: DRY kwarg passing by @larsoner in https://github.com/mne-tools/mne-bids-pipeline/pull/746
> * ENH: Add movement compensation by @larsoner in https://github.com/mne-tools/mne-bids-pipeline/pull/747
> * Fix for documentation issue related to processing EEG separately in a combined M/EEG dataset by @allermat in https://github.com/mne-tools/mne-bids-pipeline/pull/751
> * MAINT: Refactor _get_raw_paths by @larsoner in https://github.com/mne-tools/mne-bids-pipeline/pull/749
> 
> ## New Contributors
> * @allermat made their first contribution in https://github.com/mne-tools/mne-bids-pipeline/pull/743
> 
> **Full Changelog**: https://github.com/mne-tools/mne-bids-pipeline/compare/v1.3.0...v1.4.0
> 

I renamed a couple of PRs because they had rather non-descriptive titles. Please let's ensure that before merging, PR titles look like something that we'd actually want to appear in the changelog. 

@larsoner I can make the release tomorrow, just wanted to get your okay to move forward.

**Comment by larsoner** [comment]
Works for me! There's a lot more to add for movement compensation but this will at least get things started

**Comment by agramfort** [comment]
+1

**Comment by hoechenberger** [comment]
I tagged the release but the doc deployment is failing 

And we have issues in the docs themselves. Maybe we should change the settings to fail hard on warnings:

> WARNING  -  Documentation file 'changes.md' contains a link to '[mne_bids_pipeline._config.ssp_ecg_channel' which is not found in the documentation files.
WARNING  -  Documentation file 'changes.md' contains a link to '[mne_bids_pipeline._config.read_raw_bids_verbose' which is not found in the documentation files.
WARNING  -  Documentation file 'changes.md' contains a link to '[mne_bids_pipeline._config.mf_reference_run' which is not found in the documentation files.
WARNING  -  Documentation file 'changes.md' contains a link to 'mne_bids_pipeline._config.n_jobs' which is not found in the documentation files.
WARNING  -  Documentation file 'settings/preprocessing/maxfilter.md' contains a link to 'settings/preprocessing/mne_bids_pipeline._config.mf_destination' which is not found in the documentation files.

**Comment by larsoner** [comment]
I tried restarting the deployment but it failed. Looks like this is the issue:
```
remote: fatal: pack exceeds maximum allowed size (2.00 GiB)        
```
So we probably need to find a way to make our website smaller

**Comment by hoechenberger** [comment]
We could include only the reports for the first participant and grand average for each example

**Comment by larsoner** [comment]
I think ultimately keeping all doc versions forever will be untenable. Some options:

## 1. Archive and link to old doc versions

1. Keep the last N versions available as HTML -- maybe 2, which would mean keep 1.4 (just released) and 1.3?
2. Versions older than that get zipped, uploaded to OSF.io or similar, and we add a link to them somewhere in our docs.

Has the advantage of staying complete, but `mike` might not be happy with it, and it's not easy to see what config values were around in 1.0 for example.

## 2. Archive *report HTMLs* from old versions

For this, we could manually:

1. take the report HTMLs for 1.0, 1.1, and 1.2, zip each one, and upload to OSF.io
2. `rm` the HTMLs for those versions from the website
3. Add a javascript HTML banner that says "report HTML links for this unsupported version of MNE-BIDS-Pipeline will not work, but they can be downloaded here: <link>"
4. *and/or* we modify the generated website to have all report download links point to the OSF.io zip link or something

Has the advantage of being browsable, but gives the sense that report HTMLs will be readily available when they're not.

## 3. Remove and strip out report HTMLs from old versions

I think this is my preference:

1. `rm` the old report HTML files
2. modify the generated HTML to have `visibility: hidden` on the "Generated output" heading and box (either manually do it with a search/replace, or do it in JavaScript). Or we can just kill those HTML sections altogether (though this is maybe harder to script).

Has the advantage of being browseable, but the old report HTMLs are gone. This is the easiest and I don't think users will really care what the old report HTML from unsupported MNE-BIDS-Pipelines used to look like.

@hoechenberger any other ideas? If not, okay with you if I just do (3)?

**Comment by hoechenberger** [comment]
@larsoner I don't think anyone cares about the reports for old versions, so 3 is entirely fine with me

**Comment by larsoner** [comment]
Okay locally I did:
```
$ git clone --single-branch --branch gh-pages git@github.com:/mne-tools/mne-bids-pipeline-website
$ cd mne-bids-pipeline-website
$ find $PWD/1.3/examples/ -type d -name "*" | tail -n +2 | xargs rm -Rf
... repeat for other versions
$ find 1.3/examples/ -name "*.html" -exec sed -i /^\<h2\ id=\"generated-output\"\>Generated/,/^\ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ $/{d} {} \;
... repeat for other versions
$ find 1.3/examples/ -name "*.html" -exec sed -i '/^  <a href="#generated-output"/,/^  <\/a>$/{d}' {} \;
... repeat for other versions
$ git commit --amend -a
$ git push -f
```
After things looked okay locally. Deployment looks okay I think:

before | after
--|--
![Screenshot from 2023-07-05 16-07-17](https://github.com/mne-tools/mne-bids-pipeline/assets/2365790/5a3c4f82-ea51-4700-8b65-e78c002f4045) | ![Screenshot from 2023-07-05 16-10-35](https://github.com/mne-tools/mne-bids-pipeline/assets/2365790/9ac92a05-860d-4720-ad5e-fe351aa1c5e1)




We can "just" use these commands every once in a while to cull the example HTML files until we find a way to systematically delete them...

**Comment by hoechenberger** [comment]
Agh, doc deployment failing again:

"fatal: pack exceeds maximum allowed size"

for the latest merge into `main` and, more importantly, for the latest tagged release

**Comment by larsoner** [comment]
Hmm this is one reason maybe we should be doing point releases for little bugfixes, those would be deployed to the same directories. The only quick fix I can think of is to delete 1.5 now for example

**Comment by larsoner** [comment]
We should probably figure out how to get this down:
```
larsoner:~/python/mne-bids-pipeline$ du -hs 1.5
769M	1.5
larsoner:~/python/mne-bids-pipeline$ du -hs 1.6
854M	1.6
larsoner:~/python/mne-bids-pipeline$ du -hs 1.7
854M	1.7
```
and
```
$ du -hs 1.7/examples/
849M	1.7/examples/
$ du -hs 1.7/examples/ERP_CORE
510M	1.7/examples/ERP_CORE
```
So a good target might be looking at the ERP_CORE 
```
6.3M	1.7/examples/ERP_CORE/sub-019_ses-P3_task-P3_proc-ica+components_report.html
1.3M	1.7/examples/ERP_CORE/sub-019_ses-P3_task-P3_proc-icafit_report.html
1.5M	1.7/examples/ERP_CORE/sub-019_ses-P3_task-P3_proc-ica_report.html
6.5M	1.7/examples/ERP_CORE/sub-019_ses-P3_task-P3_report.html
```
So back of the envelope 15MB per subject per task, 16*5*7=525MB. Options:

1. *Run* all subjects but only upload *one subject's reports*, which seems reasonable -- that cuts this down greatly.
2. Reduce the number of tasks -- not sure what the consequences are here esp. since we do different types of analyses on different tasks.
3. Follow https://github.com/mne-tools/mne-bids-pipeline/issues/880#issuecomment-1991078989 and remove two of the ICA reports.

I'll try (1) and (3) in a PR and we can see how big the resulting doc build is.

**Comment by hoechenberger** [comment]
thanks for looking into this!

**Comment by larsoner** [comment]
All fixed and example-removing scripts added in #899 so we can hopefully stop removing old versions (forgot we could do that otherwise I wouldn't have removed the older ones!)
